src/conn.c

```diff
@@ -799,7 +799,9 @@ int xmpp_connect_component(xmpp_conn_t *conn,
         return XMPP_EINVOP;
 
     /* XEP-0114 does not support TLS */
-    xmpp_conn_disable_tls(conn);
+    long flags = xmpp_conn_get_flags(conn);
+    flags |= XMPP_CONN_FLAG_DISABLE_TLS;
+    (void)xmpp_conn_set_flags(conn, flags);
```

Comment:
Generally good PR, but I might suggest to move "sending less acks" to another PR in case if it can be potentially breaking change. Most of the changes are harmless and barely touch functionality (like memset), while about others I am not sure (though I didn't dive that deep).

Reply:
> I might suggest to move "sending less acks" to another PR in case if it can be potentially breaking change

It's already a separate commit and nobody tests PRs of this repo, so why bother? If it breaks for someone they will report and we'll fix it.

Reply:
I am unable to resolve it:
![image](https://github.com/strophe/libstrophe/assets/129467592/68b74175-9a7b-4618-af74-c73791b774e9)